### PR TITLE
Add DD mappings for locking switch

### PR DESCRIPTION
**Pull request by waffle87**
## Description

Title. Am open to this living under a different/existing key

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

File: data/mappings/info_config.hjson [hunk 1]
```diff
@@ -66,6 +66,10 @@
     "LEADER_KEY_STRICT_KEY_PROCESSING": {"info_key": "leader_key.strict_processing", "value_type": "bool"},
     "LEADER_TIMEOUT": {"info_key": "leader_key.timeout", "value_type": "int"},
 
+    // Locking Switch
+    "LOCKING_SUPPORT_ENABLE": {"info_key": "locking.enabled", "value_type": "bool"},
+    "LOCKING_RESYNC_ENABLE": {"info_key": "locking.resync", "value_type": "bool"},
+
```
Comment (fauxpark) [inline data/mappings/info_config.hjson:72]: Should go below the LED Matrix section to be in alphabetical order.

File: data/schemas/keyboard.jsonschema [hunk 2]
```diff
@@ -346,6 +346,14 @@
...
             }
         },
+        "locking": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "enabled": {"type": "boolean"},
+                "resync": {"type": "boolean"}
+            }
+        },
```
Comment (fauxpark) [inline data/schemas/keyboard.jsonschema:356]: Let's move this under `qmk`.

File: docs/reference_info_json.md [hunk 3]
```diff
@@ -370,6 +370,16 @@ Configures the [LED Matrix](feature_led_matrix.md) feature.
...
+## Mechanical Lock Switch Support :id=locking-support
+
+* `locking`
+    * `enabled`
+        * Enable locking switch support
+        * Default: `false`
+    * `resync`
+        * Keep switch state consistent with keyboard LED state
+        * Default: `false`
+
```
Comment (fauxpark) [inline docs/reference_info_json.md:382]: As above.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: docs/reference_info_json.md [hunk 4]
```diff
@@ -66,6 +66,14 @@ You can create `info.json` files at every level under `qmk_firmware/keyboards/<k
     * `tap_keycode_delay`
         * The delay between keydown and keyup for tap events in milliseconds.
         * Default: `0` (no delay)
+    * `locking`
```
Comment (fauxpark) [inline docs/reference_info_json.md:69]: Should go above `tap_capslock_delay` to be in alphabetical order.

File: docs/reference_info_json.md [hunk 5]
```diff
@@ -60,12 +60,20 @@ You can create `info.json` files at every level under `qmk_firmware/keyboards/<k
     }
     ```
 * `qmk`
+    * `locking`
+        * `enabled`
+            * Enable locking switch support
```
Comment (fauxpark) [inline docs/reference_info_json.md:65]:
```suggestion
            * Enable locking switch support.
```

File: docs/reference_info_json.md [hunk 6]
```diff
@@ -60,12 +60,20 @@ You can create `info.json` files at every level under `qmk_firmware/keyboards/<k
...
+            * Default: `false`
+        * `resync`
+            * Keep switch state consistent with keyboard LED state
```
Comment (fauxpark) [inline docs/reference_info_json.md:68]:
```suggestion
            * Keep switch state consistent with keyboard LED state.
```

File: docs/reference_info_json.md [hunk 7]
```diff
@@ -60,12 +60,20 @@ You can create `info.json` files at every level under `qmk_firmware/keyboards/<k
...
+            * Default: `false`
     * `tap_capslock_delay`
         * The delay between keydown and keyup for Caps Lock tap events in milliseconds.
         * Default: `80` (80 ms)
     * `tap_keycode_delay`
         * The delay between keydown and keyup for tap events in milliseconds.
         * Default: `0` (no delay)
+
```
Comment (fauxpark) [inline docs/reference_info_json.md:76]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**